spec/unit/application/agent_spec.rb

```diff
@@ -588,6 +588,21 @@
         execute_agent
       end
     end
+
+    describe "when starting in daemon mode" do
+      before :each do
+        allow(Puppet).to receive(:notice)
+      end
+      it "should not print config in default mode" do
+        expect(@puppetd).not_to receive(:get_config)
+        execute_agent
+      end
+      it "should print config in debug mode" do
+        @puppetd.options[:debug] = true
+        expect(@puppetd).to receive(:get_config)
```

Comment:
The expectation prevents the `get_config` method from being evaluated, so we don't know if that method actually works. Better to make an expectation about the behavior, such as we expect the output to contain a known setting and value. When running tests, all [log output is collected in an array](https://github.com/puppetlabs/puppet/blob/master/spec/spec_helper.rb#L131-L141_ and made available to the test via `@logs` instance variable. So you can do something like:

```ruby
expect(@logs).to contain(an_object_having_attributes(level: notice, message: /runinterval\s*1800/))
```

Reply:
on-going, have some troubles because config face calls $stderr.puts for some reason ..